doc/event-alarm-framework/events-producer.md

```diff
@@ -0,0 +1,212 @@
+- The rsyslog plugin binary being under host control, ensures a single/unified behavior across all.
+- The unit tests can use hardcoded log messages to validate regex.
+
+
+![image](https://user-images.githubusercontent.com/47282725/157343412-6c4a6519-c27b-459b-896b-7875d8f952b8.png)
+
+##### Pros & Cons
+
+###### Pro
+1) This support is external to app, hence adapts well with III party applications, bgp, teamd, ... 
+2) This feature can be added to released builds too, as all it takes is to copy two files into each container and restart rsyslogd in the container
+3) The regex for parsing being local to container, it supports any container upgrade transaparently.
+4) The rsyslog plugin binary is maintained by host, hence provide a overall control across all containers.
+5) The message parsing load is distributed as per container. Within a container parsing is done at the granularity of per process with no extra cost as rsyslogd already pre-parsed it per-process, *always*
+
+###### con:
+1) Two step process for devs, as for each new/updated log message ***for an event***, add/update regex as needed. Updating code directly to raise the event will help avoid this.
+2) The rsyslogd is *required*. It should be treated as critical process in each container.
+
+### Event reporting
+
+#### requirements
+- Events are reported from multiple event detectors or we may call event-sources.
+- The event detectors could be running in host and/or some/all containers.
+- Support multiple local clients to be able to receive the events concurrently.
+- Each local client should be able to receive updates from all event detectors w/o being aware of all of the sources.
+- The local clients could be operating at different speeds.
+- The clients may come and go.
+- There may not be any local client to receive the events.
+- Events reporting should not be blocked by any back pressure from any local client.
+
+#### Design
+- Event detectors send UDP messages to a multicast group
+- The local clients add themselves as members and receive from the group.
+
+##### Pro
+- The senders are never blocked.
+- The receivers can be 0 to many.
+- Both senders & receivers are neither aware of each other nor has any binding.
+- Simple design, hence more reliable.
+
+##### con
+- Messages could get lost, if the client is slow. It is the client's responsibility to ensure no loss.
+- Clients that are slow by design (*like redis-DB updater*), could have a dedicated thread to receive all events and cache the latest. The main thread could be slow, it could miss updates, but it would use/record the latest.
+
+
+### Event exporting
+
+#### requirements
+- Telemetryt container receives all locally raised events.
+- Telemetry container supports exporting all the locally raised events to one or more external clients.
+- RFE: When restarted, ensure to provide the latest on all events that were missed during downtime.
```

Comment:
>RFE [](http://example.com/codeflow?start=2&length=3)

Are you going to resolve all 'RFE' lines in this design PR?

Reply:
This will be covered by telemetry client design.